src/lib.rs

```diff
@@ -498,9 +498,32 @@ macro_rules! ImplDekuTraits {
     };
 }
 
-impl<T: DekuRead<Ctx>, Ctx: Copy> DekuRead<(Count, Ctx)> for Vec<T> {
-    /// Read the specified number of `T`s from input.
-    /// * `count` - the number of `T`s you want to read.
+fn read_vec_with_predicate<T: DekuRead<Ctx>, Ctx: Copy, Predicate: FnMut(usize, &T) -> bool>(
+    input: &BitSlice<Msb0, u8>,
+    vec: Vec<T>,
+    ctx: Ctx,
+    mut predicate: Predicate,
+) -> Result<(&BitSlice<Msb0, u8>, Vec<T>), DekuError> {
+    let mut res = vec;
+    let mut rest = input;
+    loop {
+        let (new_rest, val) = <T>::read(rest, ctx)?;
+        res.push(val);
+        rest = new_rest;
+
+        if predicate(input.offset_from(rest) as usize, res.last().unwrap()) {
```

Comment:
Handle unwrap

Reply:
I'm unsure what the best practice is for handling the unwrap here, given that it's trivially impossible for it to fail as we're pushing to the vec immediately before it

Reply:
Fair enough! :)
A quick comment saying that would help in the future when looking at possible panics